### PR TITLE
Add pre database objects definition parser changes

### PR DESCRIPTION
**Pull request by epicallan**

### CODE REVIEW

File: pgt/src/PGT/Output/Golden.hs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          (Path.toString path)
+          (Path.toString outputFile)
+          runTest
+      where
+        runTest :: IO Text
+        runTest = do
+          content <- Text.decodeUtf8 <$> BS.readFile (Path.toString inputFile)
+
+          either (pure . removeDynamicText . convert) (pure . render)
+            =<< catch @SomeException (evaluate $ parser content) (pure . Left . show)
```
Comment (epicallan) [inline pgt/src/PGT/Output/Golden.hs:37]: comment: using `evaluate` forces function call to WHF which exposes thunked errors, which is required for catching `ErrorCall` errors in lazy pure code.

File: pgt/src/PGT/Prelude.hs [hunk 2]
```diff
@@ -1,4 +1,11 @@
...
+  )
+where
 
-import Data.Conversions as Exports
-import MPrelude         as Exports
+import Data.Conversions    as Exports
+import MPrelude            as Exports
+
+showc :: forall b a . (Show a, Conversion b String) => a -> b
+showc = convert . show
```
Comment (dkubb) [inline pgt/src/PGT/Prelude.hs:14]: __question:__ Not a blocker, but this appears in  multiple projects. Should it be moved to `MPrelude` eventually?

File: pgt/src/PGT/Output/Text.hs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+parseName = do
+  head <- Text.takeWhile1 Char.isAlphaNum
+  tail <- Text.many' parseTail
+
+  pure . Text.concat $ head : tail
+  where
+    parseTail :: Parser Text
+    parseTail = do
+      char      <- parseSeparator
+      remaining <- Text.takeWhile1 Char.isAlphaNum
```
Comment (dkubb) [inline pgt/src/PGT/Output/Text.hs:47]: __suggestion: (nitpick)__ The expression `Text.takeWhile1 Char.isAlphaNum` appears here and on line 15. Should this be a private function and reused in both places?

File: pgt/src/PGT/Output/Text.hs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      where
+        parseSeparator :: Parser Char
+        parseSeparator
+          = Text.choice
+          $ Text.char <$>
+          [ '-'
+          , '.'
+          , '_'
+          , ' '
+          ]
```
Comment (dkubb) [inline pgt/src/PGT/Output/Text.hs:59]: __question:__ is the intention with this function to parse one or more alpha numeric characters, then one separator, and then one or more alpha numeric characters? Does the separator only appear one time in the name?

File: pgt/src/PGT/Output/Text.hs [hunk 5]
```diff
@@ -55,3 +56,6 @@ parseName = do
           , '_'
           , ' '
           ]
+
+unlines :: (Foldable f, Conversion Text a) => f a -> Text
```
Comment (dkubb) [inline pgt/src/PGT/Output/Text.hs:64]: __suggestion: (nitpick)__ The `Foldable` type should appear last in the set of types.

**Review by dkubb — APPROVED** [review]

**Pull request merged**